argh/src/lib.rs

```diff
@@ -677,19 +681,29 @@ pub fn parse_struct_args(
     mut parse_positionals: ParseStructPositionals<'_>,
     mut parse_subcommand: Option<ParseStructSubCommand<'_>>,
     help_func: &dyn Fn() -> String,
+    help_json_func: &dyn Fn() -> String,
 ) -> Result<(), EarlyExit> {
     let mut help = false;
+    let mut help_json = false;
     let mut remaining_args = args;
     let mut positional_index = 0;
     let mut options_ended = false;
 
     'parse_args: while let Some(&next_arg) = remaining_args.get(0) {
         remaining_args = &remaining_args[1..];
+
         if (next_arg == "--help" || next_arg == "help") && !options_ended {
             help = true;
             continue;
         }
 
+        // look for help-json for json formatted help output.
+        if (next_arg == "--help-json" || next_arg == "help-json") && !options_ended {
```

Comment:
Good point regarding backwards compatibility.
I'll take a look to try to understand your idea :) (This is driving me to learn Rust *much* better). If you get something more fleshed out, let me know. I am happy to pick it up and flesh it out.

Reply:
Two ideas that came to mind, looking for thoughts:

1. add a field to [EarlyExit](https://github.com/google/argh/blob/master/argh/src/lib.rs#L462) : `output_json` which has the json encoded help, or in the case of other errors, a json encoded version of `{"output": "the message"}`.
2. add a method to argh::help_json(). This is similar to #56 so applications could explicitly access the help message.

Reply:
I may misunderstand your suggestion on (1) but generating JSON help content just to throw an error sounds like a lot of unnecessary overhead. Most use cases wouldn't need to process the JSON output, I'm thinking. JSON encoding should be on-demand, I think.

Reply:
My original idea was to follow the `--help` flag, so if the help was generate and returned, but then it seemed like it would be strange on errors to have "output" and not "output_json". 

But based on your comment, I am leaning more towards the explicit method.

Reply:
Yeah I think going with a dedicated method would be best. That’d be easier for the optimizer to remove if unused.

One of the troubles we ran into with clap was that it carried around a lot of extra state to do some fancy stuff like typo suggestions, and it was hard for the optimizer to reliably remove that state, even if it was unused.

Reply:
Added the method and a default implementation so non-derived implementations will be backwards compatible.